test/parallel/market.rpc.check.test.ts

```diff
@@ -0,0 +1,473 @@
+  threshold = stringToBN(
+    JSON.parse(JSON.stringify(meta)).swapValueThreshold.toString(),
+  );
+  sudo = getSudoUser();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.FinalizeTge(),
+    Assets.initIssuance(),
+    Assets.mintNative(sudo),
+  );
+});
+
+beforeEach(async () => {
+  [testUser] = setupUsers();
+
+  [firstCurrency, secondCurrency] = await Assets.setupUserWithCurrencies(
+    sudo,
+    [threshold.muln(100), threshold.muln(100)],
+    sudo,
+  );
+});
+
+describe.each(["Xyk", "StableSwap"])(
+  "Check RPC calculate methods for % pools",
+  (poolType) => {
+    beforeEach(async () => {
+      await Sudo.batchAsSudoFinalized(
+        Market.createPool(
+          firstCurrency,
+          threshold.muln(10),
+          secondCurrency,
+          threshold.muln(5),
+          poolType,
+        ),
+        Assets.mintNative(testUser),
+        Assets.mintToken(firstCurrency, testUser, threshold.muln(20)),
+      );
+
+      await updateFeeLockMetadata(sudo, null, null, null, [
+        [firstCurrency, true],
+      ]);
+
+      liqId = await getPoolId(firstCurrency, secondCurrency);
+    });
+
+    test("Function calculateSellPriceByMarket works correctly", async () => {
+      const sellPrice = await calculateSellPriceByMarket(
+        liqId,
+        firstCurrency,
+        threshold.muln(2),
+      );
+
+      if (poolType === "Xyk") {
+        const poolBalance = await getBalanceOfPool(
+          firstCurrency,
+          secondCurrency,
+        );
+        priceCalculationRetro = await calculate_sell_price_rpc(
+          poolBalance[0],
+          poolBalance[1],
+          threshold.muln(2),
+        );
+      } else {
+        priceCalculationRetro = sellPrice;
+      }
+
+      //we create reverse calculation to check that we receive another value
+      const sellPriceReverse = await calculateSellPriceByMarket(
+        liqId,
+        secondCurrency,
+        threshold.muln(2),
+      );
+
+      await signTx(
+        api,
+        Market.sellAsset(
+          liqId,
+          firstCurrency,
+          secondCurrency,
+          threshold.muln(2),
+        ),
+        testUser.keyRingPair,
+      ).then((result) => {
+        const eventResponse = getEventResultFromMangataTx(result);
+        expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      const tokenAmount = await getTokensAccountInfo(
+        testUser.keyRingPair.address,
+        secondCurrency,
+      );
+
+      expect(sellPrice).bnEqual(priceCalculationRetro);
+      expect(sellPrice).not.bnEqual(sellPriceReverse);
+      expect(stringToBN(tokenAmount.free)).bnEqual(sellPrice);
+    });
+
+    test("Function calculateSellPriceWithImpact makes reliable calculations", async () => {
+      const sellPriceBefore = await calculateSellPriceByMarket(
+        liqId,
+        firstCurrency,
+        threshold,
+      );
+
+      const sellPriceWithImpact = await calculateSellPriceWithImpact(
```

Comment:
what are [0] and [1] ?

Reply:
I need a comment here, I don't understand